### PR TITLE
feat: added a new helper-api-error package

### LINKED ISSUE #325: Incorrect cross-package dependencies
Many of the packages depend on `webassemblyjs`, which results in them indirectly depending on everything else.

### PR DESCRIPTION
**Pull request by ColinEberhardt**
replaces #344 
fixes #325

### CODE REVIEW

File: packages/wasm-parser/src/decoder.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // @flow
 
-import { CompileError } from "webassemblyjs/lib/errors";
+const { CompileError } = require("@webassemblyjs/helper-api-error");
```
Comment (xtuc) [inline packages/wasm-parser/src/decoder.js]: can you use ES6 import here?

File: packages/wast-parser/src/number-literals.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // @flow
 const Long = require("long");
 const parseHexFloat = require("@webassemblyjs/floating-point-hex-parser");
-const { CompileError } = require("webassemblyjs/lib/errors");
+const { CompileError } = require("@webassemblyjs/helper-api-error");
```
Comment (xtuc) [inline packages/wast-parser/src/number-literals.js]: same, can you use ES6 imports please

File: packages/wasm-parser/src/decoder.js (hunk 1 above)
Comment (ColinEberhardt) [reply packages/wasm-parser/src/decoder.js]: Oops - good point. Fixed that

**Comment by ColinEberhardt** [comment]
> Cool, that's also a good first step to remove our cyclic deps
indeed - only a few left now.

**Pull request merged**